carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -959,11 +988,25 @@ public boolean isElementWithTextPresent(final String text, long timeout) {
     	final String decryptedText = cryptoTool.decryptByPattern(text, CRYPTO_PATTERN);
 		ExpectedCondition<Boolean> textCondition;
 		if (element != null) {
+			ExpectedCondition<Boolean>  tmpCondition = ExpectedConditions.and(ExpectedConditions.visibilityOf(element));
+			boolean tmpResult = waitUntil(tmpCondition, 0);
+			
+			if (!tmpResult && originalException != null && StaleElementReferenceException.class.equals(originalException.getClass())) {
+				LOGGER.debug("StaleElementReferenceException detected in isElementWithTextPresent!");
+				try {
+					refindElement();
+					textCondition = ExpectedConditions.textToBePresentInElement(element, decryptedText);
+				} catch (NoSuchElementException e) {
+					// search element based on By if exception was thrown
+					textCondition = ExpectedConditions.textToBePresentInElementLocated(getBy(), decryptedText);
+				}
+			}
+			
 			textCondition = ExpectedConditions.textToBePresentInElement(element, decryptedText);
 		} else {
 			textCondition = ExpectedConditions.textToBePresentInElementLocated(getBy(), decryptedText);
 		}
-		return waitUntil(ExpectedConditions.and(getDefaultCondition(getBy()), textCondition), timeout);
+		return waitUntil(textCondition, timeout);
     	//TODO: restore below code as only projects are migrated to "isElementWithContainTextPresent"
 //    	return waitUntil(ExpectedConditions.and(ExpectedConditions.presenceOfElementLocated(getBy()),
 //				ExpectedConditions.textToBe(getBy(), decryptedText)), timeout);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)